src/dune_rules/merlin/ocaml_index.ml

```diff
@@ -32,11 +32,9 @@ let cctx_rules cctx =
 
            The implicit transitive libs correspond to the set:
            (requires_link \ req_compile) *)
-        let* req_link = Compilation_context.requires_link cctx in
-        let+ req_compile = Compilation_context.requires_compile cctx in
-        List.filter req_link ~f:(fun l -> not (List.exists req_compile ~f:(Lib.equal l)))
+        Compilation_context.requires_hidden cctx
```

Comment:
This doesn't seem quite right. For new versions of OCaml (>= 5.2) and version of the Dune language >= 3.17, Dune will pass `-H` to the compiler and these flags will be recorded in the `.cmt` files and will, in turn, be read by `ocaml-index`. Accordingly, in these cases we do not need to do anything special here. It is only for backwards compatibility that we can keep the above logic (ie if OCaml is < 5.2 _or_ if the Dune language veresion if < 3.17). Also, the flags should all be passed with `-I`, not `-H` in this case.

Reply:
@voodoos sounds OK?

Reply:
Yes, that's completely right. If Dune >= 3.17 there is no more additional flags to pass to the indexer. (Note that indexing is not available prior to OCaml 5.2, so there is no need to check that here.)

Reply:
Thanks for the confirmation! So only the check for OCaml lang version < 3.17 is needed.

Reply:
Done!